sunpy/net/scraper.py

```diff
@@ -436,15 +370,19 @@ def _extract_files_meta(self, timerange, extractor, matcher=None):
         `list` of `dict`
             List of metadata info for all URLs.
         """
-        self.extractor = extractor
         urls = self.filelist(timerange)
         metalist = []
         for url in urls:
-            metadict = parse(extractor, url)
+            metadict = parse(self.pattern, url)
             if metadict is not None:
                 append = True
                 metadict = metadict.named
                 metadict['url'] = url
+                if 'month' not in metadict:
+                    if 'month_name' in metadict:
+                        metadict['month'] = datetime.strptime(metadict['month_name'], '%B').month
+                    elif 'month_name_abbr' in metadict:
+                        metadict['month'] = datetime.strptime(metadict['month_name_abbr'], '%b').month
```

Comment:
Is there a test to cover these lines of code?

Reply:
yes, i included that like [this](https://github.com/sunpy/sunpy/pull/7077/files#diff-616d8052c8d92faa18682a0257432d6b0dfbd390d6efab4b4e55a8abca94871fR283-R287)